server.py

```diff
@@ -18,4 +18,7 @@
 if os.name == 'nt':
     subprocess.run(['waitress-serve', '--port=3001', 'bot_python_sdk.api:api'])
 else:
-    subprocess.run(['gunicorn', '-b', '127.0.0.1:3001', 'bot_python_sdk.api:api'])
+    cmd = subprocess.Popen(['hostname', '-I'], stdout=subprocess.PIPE)
+    IPAddr = cmd.communicate()[0].decode('ascii').split(' ')[0] 
+    print("Detected IP Address :" + IPAddr)
+    subprocess.run(['gunicorn', '-b', IPAddr+':3001', 'bot_python_sdk.api:api'])
```

Comment:
Always there should be something at index 0. Added logic to validate the ip address before considering it otherwise loop back address is considered to start the server. Used Logger to print as info messages